odoo/addons/base/ir/ir_mail_server.py

```diff
@@ -464,6 +465,70 @@ def send_email(self, message, mail_server_id=None, smtp_server=None, smtp_port=N
             raise MailDeliveryException(_("Mail Delivery Failed"), msg)
         return message_id
 
+    @api.multi
+    def _get_canonical_from(self, original_from):
+        """Returns a FROM header to respect the SMTP server settings.
+
+        Args:
+            original_from (str): The original FROM header for the message.
+
+        Returns:
+            str: A string that should be used in a message header.
+        """
+        # Use the canonical name, if existing.
+        address = parseaddr(original_from)
+        # Otherwise we use the name part of the email address.
+        if not address[0]:
+            address[0] = address[1].split('@', 1)[0]
+        return '%s via %s <%s>' % (
```

Comment:
I think "_via_" will make sense in many languages, but we might still want to make it translatable. We don't know the language of the recipient at this point, but it makes sense to use the user's language (same way "Re:" and "Fwd:" are typically handled)

To help translators get the meaning of this out-of-context word, we might want to rewrite the string format with named placeholders, and remove as much technical details as possible, e.g.:

```python
from = _("%(author)s via %(company)s") % from_params
from = '"%s" <%s>' % (from, self._get_default_bounce_address())
```

Reply:
Made this change with a slight alteration. Was there a reason for encapsulating the string format for the name part in quotes (`"%s"`)?

I changed to:

```
'%s <%s>' % (str_name, self._get_default_bounce_address())
```

Reply:
Double quoting is required in when the name contains special characters.
We should use the helper function
```python
email.utils.formataddr((str_name, self._get_default_bounce_address()))
```

Reply:
Thanks for the clarification @KangOl